sdk/eventhub/event-hubs/src/models/public.ts

```diff
@@ -128,6 +128,61 @@ export interface EventHubClientOptions {
   userAgent?: string;
 }
 
+/**
+ * An options bag to configure load balancing settings.
+ * - `loadBalancingOptions`: Options to tune how the EventHubConsumerClient claims partitions.
+ * - `userAgent`        : A string to append to the built in user agent string that is passed as a connection property
+ * to the service.
+ * - `webSocketOptions` : Options to configure the channelling of the AMQP connection over Web Sockets.
+ *    - `websocket`     : The WebSocket constructor used to create an AMQP connection if you choose to make the connection
+ * over a WebSocket.
+ *    - `webSocketConstructorOptions` : Options to pass to the Websocket constructor when you choose to make the connection
+ * over a WebSocket.
+ * - `retryOptions`     : The retry options for all the operations on the EventHubConsumerClient.
+ * A simple usage can be `{ "maxRetries": 4 }`.
+ *
+ * Example usage:
+ * ```js
+ * {
+ *     retryOptions: {
+ *         maxRetries: 4
+ *     }
+ * }
+ * ```
+ */
+export interface EventHubConsumerClientOptions extends EventHubClientOptions {
+  /**
+   * Options to tune how the EventHubConsumerClient claims partitions.
+   */
+  loadBalancingOptions?: LoadBalancingOptions;
+}
+
+/**
+ * An options bag to configure load balancing settings.
+ */
+export interface LoadBalancingOptions {
+  /**
+   * Whether to apply a greedy or a more balanced approach when
+   * claiming partitions.
+   *
+   * This option is ignored when either:
+   *   - `CheckpointStore` is __not__ provided to the `EventHubConsumerClient`.
+   *   - `subscribe()` is called for a single partition.
+   * Default: balanced
+   */
+  strategy?: "balanced" | "greedy";
```

Comment:
How does the user get from here to the explanation of the different algorithms?

Reply:
Good call out, I definitely need more documentation around this.

Reply:
I 'borrowed' some text from .NET around the greedy and balanced strategies.